### DISCUSSION TITLE
Reverse scored likert

### DISCUSSION THREAD

**Discussion by viviers** [jspsych/jsPsych#1032] [Q&A] [answered]
I'm creating a trial for the Need for Cognition Scale (NCS). 

Some of the items are reverse scored but I don't know if there's a way to do this in jsPsych. 

I found this online: https://osf.io/xaf8e/ which seems to be similar to the survey-likert plugin, but it has reverse scoring. 

Would I just be able to download that .js file & put it in my plugins folder?

**Comment by jodeleeuw** [comment, marked as answer]
> Would I just be able to download that .js file & put it in my plugins folder?

Probably. It may depend a bit on what version of jsPsych the plugin is compatible with, but in theory plugins should work like this.

You can also reverse score items by using a custom function in the `on_finish` event for the trial. 

```js
on_finish: function(data){
	// convert responses from JSON string to JS object
	var responses = JSON.parse(data.responses);

	// reverse score the first question, assuming a 5-point scale
	responses.Q0 = 4 - responses.Q0;

	// save the updated data
	data.responses = JSON.stringify(responses);
}
```

Here's a fuller example in context:
```js
// Variable with the likert scale used
var response_scale = ['Not at all', 'Somewhat', 'Moderately', 'Very Much'];

// Behavioral about response to vignette
var likert = {
    type: 'survey-likert',
    questions: [
        {
          prompt: "I feel calm",
          labels: response_scale,
        },
        {
          prompt: "I am tense",
          labels: response_scale,
        },
        {
          prompt: "I feel upset",
          labels: response_scale,
        }
    ],
    scale_width: 500,
    on_finish: function(data){
      var responses = JSON.parse(data.responses);
      
      responses.Q0 = 3 - responses.Q0;
      
      data.responses = JSON.stringify(responses);
    }
};

jsPsych.init({timeline: [likert],
  on_finish: function(){
    jsPsych.data.displayData()
  }
})
```

**Reply by viviers** [reply]
You're amazing thank you so much!
